From d9d019d674f95509b5001f4d878ae09e32ea7a10 Mon Sep 17 00:00:00 2001
From: Adhemerval Zanella <adhemerval.zanella@linaro.org>
Date: Thu, 4 Jul 2024 10:09:07 -0300
Subject: [PATCH] elf: Make dl-rseq-symbols Linux only

And avoid a Hurd build failures.

Checked on x86_64-linux-gnu.

(cherry picked from commit 9fc639f654dc004736836613be703e6bed0c36a8)
---
 elf/Makefile                                       | 1 -
 sysdeps/unix/sysv/linux/Makefile                   | 4 ++++
 {elf => sysdeps/unix/sysv/linux}/dl-rseq-symbols.S | 0
 3 files changed, 4 insertions(+), 1 deletion(-)
 rename {elf => sysdeps/unix/sysv/linux}/dl-rseq-symbols.S (100%)

diff --git a/elf/Makefile b/elf/Makefile
index 04e1d7ded5..c2af11b92c 100644
--- a/elf/Makefile
+++ b/elf/Makefile
@@ -74,7 +74,6 @@ dl-routines = \
   dl-printf \
   dl-profile \
   dl-reloc \
-  dl-rseq-symbols \
   dl-runtime \
   dl-scope \
   dl-setup_hash \
diff --git a/sysdeps/unix/sysv/linux/Makefile b/sysdeps/unix/sysv/linux/Makefile
index be801e3be4..623a7d4de0 100644
--- a/sysdeps/unix/sysv/linux/Makefile
+++ b/sysdeps/unix/sysv/linux/Makefile
@@ -587,6 +587,10 @@ tests += \
 endif
 
 ifeq ($(subdir),elf)
+dl-routines += \
+  dl-rseq-symbols \
+  # dl-routines
+
 sysdep-rtld-routines += \
   dl-brk \
   dl-getcwd \
diff --git a/elf/dl-rseq-symbols.S b/sysdeps/unix/sysv/linux/dl-rseq-symbols.S
similarity index 100%
rename from elf/dl-rseq-symbols.S
rename to sysdeps/unix/sysv/linux/dl-rseq-symbols.S
-- 
2.45.0

